test/client/client.go

```diff
@@ -44,83 +46,124 @@ var (
 
 func init() {
 	flag.IntVar(&count, "count", 1, "Number of times to make the request")
-	flag.BoolVar(&parallel, "parallel", true, "Run requests in parallel")
-	flag.BoolVar(&insecure, "insecure", false, "Insecure requests")
-	flag.DurationVar(&timeout, "timeout", 60*time.Second, "Request timeout")
+	flag.DurationVar(&timeout, "timeout", 15*time.Second, "Request timeout")
 	flag.StringVar(&url, "url", "", "Specify URL")
 	flag.StringVar(&headerKey, "key", "", "Header key")
 	flag.StringVar(&headerVal, "val", "", "Header value")
 }
 
-func makeRequest(i int) func() error {
-	return func() error {
-		req, err := http.NewRequest("GET", url, nil)
-		if err != nil {
-			return err
-		}
+func makeHTTPRequest(client *http.Client) func(int) func() error {
+	return func(i int) func() error {
```

Comment:
Deleted my original comment because it was wrong. But I agree that this is a bit difficult to parse with the two levels of inline functions, though easier than the equivalent in C :)

Would it work if the XXXRequest functions were normal functions and the code at 161 simplified wrapped that in a function, e.g.
 
```golang
g.Go(func() { 
    makeHTTPRequest(client, iter) 
})
```